docs/codeql/codeql-for-visual-studio-code/using-the-codeql-model-editor.rst

```diff
@@ -0,0 +1,139 @@
+:tocdepth: 1
+
+.. _using-the-codeql-model-editor:
+
+Using the CodeQL model editor
+=============================
+
+.. include:: ../reusables/beta-note-model-pack-editor-vsc.rst
+
+You can view, write, and edit all types of CodeQL packs in Visual Studio Code using the CodeQL extension. The model editor is designed to help you model external dependencies of your codebase that are not supported by the standard CodeQL Libraries.
+
+About the CodeQL model editor
+-----------------------------
+
+The CodeQL model editor guides you through modeling the calls to external dependencies in your application or fully modeling all the public entry and exit points in an external dependency.
+
+When you open the model editor, it analyzes the currently selected CodeQL database and identifies where the application uses external APIs and all public methods. An external (or third party) API is any API that is not part of the CodeQL database you have selected.
+
+The model editor has two different modes:
+
+- Application mode (default view): The editor lists each external framework used by the selected CodeQL database. When you expand a framework, a list of all calls to and from the external API is shown with the options available to model dataflow through each call. This mode is most useful for improving the CodeQL results for the specific codebase.
+
+- Dependency mode: The editor identifies all of the publicly accessible APIs in the selected CodeQL database. This view guides you through modeling each public API that the codebase makes available. When you have finished modeling the entire API, you can save the model and use it to improve the CodeQL analysis for all codebases that use the dependency.
+
+Displaying the CodeQL model editor
+----------------------------------
+
+#. Open your CodeQL workspace in VS Code, for example, the vscode-codeql-starter workspace.
+   If you haven't updated the ``ql`` submodule for a while, update it from ``main`` to ensure that you have the queries used to gather data for the model editor.
+#. Open the CodeQL extension and select the CodeQL database that you want to model from the "Databases" section of the left side pane.
+#. Use the command palette to run the “CodeQL: Open Model Editor (Beta)” command.
+#. The CodeQL model editor will open in a new tab and run a series of telemetry queries to identify APIs in the code.
+#. When the queries are complete, the APIs that have been identified are shown in the editor.
+
+Modeling the calls your codebase makes to external APIs
+-------------------------------------------------------
+
+You typically use this method when you are looking at a specific codebase where you want to improve the precision of CodeQL results. This is usually when the codebase uses frameworks or libraries that are not supported by CodeQL but they are not used by other teams in your organization. 
```

Comment:
`but they are not used by other teams in your organization.` -> `and if the sourcecode of the framework or library is not included in the analysis`.
One could imagine a non-monolith architecture where a project has dependencies to internal packages, but where the source code of these are not included in the analysis.